tests/MongoDB.Driver.Tests/ClientEncryptionTests.cs

```diff
@@ -0,0 +1,243 @@
+
+        private const string SchemaMap = @"{
+            ""db.coll"": {
+            ""bsonType"": ""object"",
+            ""properties"": {
+                ""encrypted_placeholder"": {
+                    ""encrypt"": {
+                        ""keyId"": ""/placeholder"",
+                        ""bsonType"": ""string"",
+                        ""algorithm"": ""AEAD_AES_256_CBC_HMAC_SHA_512-Random""
+                        }
+                    }
+                }
+            }
+        }";
+
+        // todo: add async part
+        [SkippableTheory]
+        [ParameterAttributeData]
+        public void CreateDataKeyAndDoubleEncryptionForLocalTest([Values(false)] bool async)
+        {
+            RequireServer.Check().Supports(Feature.ClientSideEncryption);
+
+            ConfigureClients(out var client, out var clientEncrypted, out var clientEncryption);
+            using (client)
+            using (clientEncrypted)
+            {
+                var datakeyOptions = new DataKeyOptions(
+                    keyAltNames: new Optional<IReadOnlyList<string>>(new[] {"local_altname"}));
+
+                var localDataKey = clientEncryption.CreateDataKey("local", datakeyOptions, CancellationToken.None);
+                localDataKey.Should().NotBeNull();
+                localDataKey.AsBsonBinaryData.GuidRepresentation.Should().Be(GuidRepresentation.Standard);
+
+                var keyVaultCollection = client
+                    .GetDatabase(__keyVaultCollectionNamespace.DatabaseNamespace.DatabaseName)
+                    .GetCollection<BsonDocument>(__keyVaultCollectionNamespace.CollectionName);
+                var keyVaultDocument = keyVaultCollection
+                    .FindSync(document => document["_id"] == localDataKey)
+                    .Single();
+                keyVaultDocument["masterKey"]["provider"].Should().Be(BsonValue.Create("local"));
+
+                var encryptOptions = new EncryptOptions(
+                    Alogrithm.AEAD_AES_256_CBC_HMAC_SHA_512_Deterministic.ToString(),
+                    keyId: localDataKey.AsBsonBinaryData.Bytes);
+                var localEncrypted = clientEncryption.Encrypt(
+                    "hello local",
+                    encryptOptions,
+                    CancellationToken.None);
+
+                var coll = clientEncrypted
+                    .GetDatabase(__collCollectionNamespace.DatabaseNamespace.DatabaseName)
+                    .GetCollection<BsonDocument>(__collCollectionNamespace.CollectionName);
+                coll.InsertOne(
+                    new BsonDocument
+                    {
+                        {"_id", "local"},
+                        {"value", localEncrypted}
+                    });
+
+                //todo: doesn't work yet
+                //var res = coll.FindSync(document => document["_id"] == "local" && document["value"] == "hello local").Single();
```

Comment:
This test doesn't work because of `Mongocryptd` doesn't encrypt `value` at all. 
So we make filtering by the unencrypted string.
Should be investigated.